android-components/components/feature/search/src/main/java/mozilla/components/feature/search/internal/SearchUrlBuilder.kt

```diff
@@ -44,15 +46,21 @@ internal class SearchUrlBuilder(
 
     private fun buildUrl(template: String, searchTerms: String): String {
         val templateUri = Uri.decode(template)
-        val urlWithSubstitutions = paramSubstitution(templateUri, Uri.encode(searchTerms))
+        val inputEncoding = searchEngine.inputEncoding ?: "UTF-8"
```

Comment:
nit: instead of checking for null here, we could make `inputEncoding` non-null and have it default to "UTF-8"